app/liquid_tags/codepen_tag.rb

```diff
@@ -1,20 +1,22 @@
 class CodepenTag < LiquidTagBase
   PARTIAL = "liquids/codepen".freeze
   REGISTRY_REGEXP =
-    %r{\A(http|https)://(codepen\.io|codepen\.io/team)/[a-zA-Z0-9_\-]{1,30}/pen/([a-zA-Z0-9]{5,32})/{0,1}\z}
+    %r{\A(http|https)://(codepen\.io|codepen\.io/team)/[a-zA-Z0-9_\-]{1,30}/(pen|pen/preview)/([a-zA-Z0-9]{5,32})/{0,1}\z}
 
   def initialize(_tag_name, link, _parse_context)
     super
     @link = parse_link(link)
```

Comment:
To handle any possible escaped characters in the link (eg `%2C`), I think we should apply `CGI.unescape_html` (similar to how it's done [here](https://github.com/forem/forem/blob/71dcc7d53f5e069debc8100edcf85fb9d5b48ba8/app/liquid_tags/codesandbox_tag.rb#L13)).

Reply:
added this here: https://github.com/forem/forem/pull/16102/files#diff-1bbc1700483e366b249f440ff13cc97f8a82a94a334d12693d53ad91b4ab2e65R8